tools/zbdsniff

```diff
@@ -1,149 +1,110 @@
+                    if nwk.source in addrMap:
+                        data = raw(p.getlayer("ZigbeeAppDataPayload"))
+                        scf = (ord(data[2]) & ~ ZBEE_SEC_ENC_MIC_32) | ZBEE_SEC_ENC_MIC_32
+                        a = data[0:2] + chr(scf) + data[3:7]
+                        c = data[7:-4]
+                        mic = data[-4:]
+                        nonce = addrMap[nwk.source] + data[3:7] + chr(scf)
+                        decrypted, success = decrypt_ccm(keyHash, nonce, mic, c, a)
+                        if success:
+                            print("[+] Decrypted:")
+                            if ord(decrypted[0]) < len(APS_CMD):
+                                print("    APS Command: " + APS_CMD[ord(decrypted[0])])
+                            if ord(decrypted[1]) < len(KEY_TYPE):
+                                print("    Key Type: " + KEY_TYPE[ord(decrypted[1])])
+                            print("    Value: " + ":".join("{:02x}".format(ord(ch)) for ch in decrypted[2:18]))
+                        else:
+                            print("[-] Decrypt failed - Wrong Key ???")
+                    else:
+                        print("[-] There was no ext_src mapping for: ", str(nwk.source))
+                else:
+                    print("[-] Unexpected key_type", str(sec.key_type))
+
+
+if __name__ == '__main__':
+    # Define the command line options.
+    parser = OptionParser(description="zbdsniff: Decode plaintext Zigbee Nwtwork key from a " +
+        "capture file. Will process libpcap or Daintree SNA capture files. Original concept: " +
+        "jwright@willhackforsushi.com, re-implemented using Scapy by Steve Martin.")
+    parser.add_option("-f", "--file", dest="filename",
+                    help="PCap file to process", metavar="FILE")
+    parser.add_option("-d", "--dir", dest="directory",
+                    help="Directory of PCap files to process", metavar="DIR")
+    parser.add_option("-k", "--transport-key", dest="transportKey",
+                    help="Transport Key for decryption")
+    parser.add_option("-v", "--verbose",
+                    action="store_true", dest="verbose", default=False,
+                    help="Print detailed status messages to stdout")
+
+    (options, args) = parser.parse_args()
+
+    if (not options.filename and not options.directory):
+        print("A packet capture file or directory must be specified")
+        sys.exit(1)
+
+    if (not options.transportKey):
+        print("A transport key value must be specified")
+        sys.exit(1)
+
+    files = []
+    if options.filename:
+        files.append(options.filename)
+
+    if options.directory:
+        files += glob.glob(os.path.join(options.directory, "*.pcap"))
+
+    filecount = 0
+
+    for file in files:
+        print "Processing %s"%file
+        if not os.path.exists(str(file)):
+            print >>sys.stderr, "ERROR: Input file \"%s\" does not exist." % file
```

Comment:
develop doesn't require py3 yet, but we should do py3 for new development to support a later transition. However, I will handle this in the py3 branch.